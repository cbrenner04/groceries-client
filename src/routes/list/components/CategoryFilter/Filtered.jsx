import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const Filtered = ({ filter, handleClearFilter }) => (
  <>
    <span id="filter-title">Filtering by:</span>
    <Button data-test-id="clear-filter" variant="outline-primary" id="filter-button" onClick={handleClearFilter}>
      {filter} <i className="fa fa-trash" />
    </Button>
  </>
);

Filtered.propTypes = {
  filter: PropTypes.string.isRequired,
  handleClearFilter: PropTypes.func.isRequired,
};

export default Filtered;
