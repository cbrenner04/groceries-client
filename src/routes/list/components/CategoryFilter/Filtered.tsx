import React, { type MouseEventHandler } from 'react';
import { Button } from 'react-bootstrap';

export interface IFilteredProps {
  filter: string;
  handleClearFilter: MouseEventHandler;
}

const Filtered: React.FC<IFilteredProps> = (props): React.JSX.Element => (
  <React.Fragment>
    <span id="filter-title">Filtering by:</span>
    <Button data-test-id="clear-filter" variant="outline-primary" id="filter-button" onClick={props.handleClearFilter}>
      {props.filter} <i className="fa fa-trash" />
    </Button>
  </React.Fragment>
);

export default Filtered;
