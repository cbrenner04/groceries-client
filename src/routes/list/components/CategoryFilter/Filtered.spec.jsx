import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Filtered from './Filtered';

describe('Filtered', () => {
  const props = {
    filter: 'foo',
    handleClearFilter: jest.fn(),
  };

  it('renders with filter name in button', () => {
    const { container, getByRole } = render(<Filtered {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByRole('button')).toHaveTextContent('foo');
  });

  it('calls handleClearFilter when button is clicked', () => {
    const { getByRole } = render(<Filtered {...props} />);

    fireEvent.click(getByRole('button'));

    expect(props.handleClearFilter).toHaveBeenCalled();
  });
});
