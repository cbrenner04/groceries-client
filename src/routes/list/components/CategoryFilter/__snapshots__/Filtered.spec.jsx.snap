// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filtered renders with filter name in button 1`] = `
<div>
  <span
    style="line-height: 2.5rem; margin-right: 1rem;"
  >
    Filtering by:
  </span>
  <button
    class="btn btn-outline-primary"
    id="clear-filter-button"
    style="margin-right: 1rem;"
    type="button"
  >
    foo
     
    <i
      class="fa fa-trash"
    />
  </button>
</div>
`;
